### ISSUE TITLE
Fallout 4 support

### ISSUE THREAD

**Issue by Utumno** [wrye-bash/wrye-bash#245]
I would like Bash to have Fallout 4 support from day one - I hope that this is as simple as plugging in a correct `game/fallout4/__init__.py` @lojack5 any tips ?

@Sharlikran what you say ?

I do not intend to install fallout so I need some help here

It is a nice opportunity to move patcher code to `game/<game>/patcher/__init__.py` and rewrite `game.template` once and for all - will help in all patcher related stuff.

**Comment by Arthmoor** [comment]
Day 1 support seems awfully ambitious. I can guarantee that the record support will simply not be adequate even if they base a lot of it on what you could do in Skyrim. I don't think it's realistic to expect records based tools to have any kind of useful support until at least 2-3 months after release.

I do agree though that it's as valid an excuse as any to isolate the patcher and game template code so that new games can be added to the program without too much trouble.

**Comment by lojack5** [comment]
Early support for BAIN  should be very easy to implement.  Early mod reading support - it really depends on how much they change up the plugin format.  Assuming minor changes, like from Oblivion to Skyrim, it should be easy to get very early support for reading the TES4 record (again assuming they keep with that format). Patchers, those will take longer, especially since we really don't even have Skyrim's implemented yet.

I will be getting Fallout 4, so I'll probably work on at least the first basic stuff needed for it, but still very busy so I probably won't go much beyond the basics needed for making Bash run and do basic things.

@WrinklyNinja will have to do similar for liblo I'm assuming, though I'd guess they're going to stick with the plugins.txt method of load ordering.

**Comment by Sharlikran** [comment]
Everything hinges on system requirements.  I have an LGA 775 based Quad Core.  So if Fallout 4 requires an i5 and a 64 bit OS, which I suspect will be the case, then I will not have the minimum system requirements and I won't be able to even install it.

**Comment by WrinklyNinja** [comment]
> @WrinklyNinja will have to do similar for liblo I'm assuming, though I'd guess they're going to stick with the plugins.txt method of load ordering.

Yes, if there's no major changes then I will be updating LOOT to support it, and therefore libloadorder too. The modding tools aren't going to be out until the new year IIRC, so there's no rush (though I anticipate TESVSnip-like events repeating themselves).

Fallout doesn't interest me otherwise though, so I'll be relying on info from those who do buy the game, or for someone to gift me it, as has been the case for the previous Fallout games.

**Comment by Arthmoor** [comment]
Minimum specs:

Windows 7/8/10 (64-bit OS required)
Intel Core i5-2300 2.8 GHz/AMD Phenom II X4 945 3.0 GHz or equivalent
8 GB RAM
30 GB free HDD space
NVIDIA GTX 550 Ti 2GB/AMD Radeon HD 7870 2GB or equivalent

Dunno if that helps or not. My rig meets them easily, as well as blowing out the recommended specs. I don't know if I'm going to pick the game up right away though either. I may end up waiting for the GOTY version before bothering, especially if they still use worldspaced cities in a game that has no reason to keep doing that.

GECK won't be out until "early 2016" which might mean as late as March. So you can bet your ass there will be terrible repeats of the Snip-era crap as people get impatient.

**Comment by Sharlikran** [comment]
So yeah, unless I suddenly get donations to buy a an i5 Motherborad, processor, and all the stuff I would need since all my hardware would be incompatible, I won't be playing or installing Fallout 4.

**Comment by Utumno** [comment]
> I will be getting Fallout 4, so I'll probably work on at least the first basic stuff needed for it, but still very busy so I probably won't go much beyond the basics needed for making Bash run and do basic things.

That's great news for me. It'd be an opportunity to also see what goes where in the game package which is one of my goals for 307 refactoring - package layout should freeze in 307

**Comment by Metallicow** [comment]
The OrangeBox could handle it, but I am on a i5-2450M at 2.50GHz laptop atm with a small HDD and Skyrim, and have my time tied with SourceCoder and modding TES stuff. Seems like everything is Steam now anyways, so I won't probably run out and buy it first day just because someone on the forums says it is a old-school install disk.
On this issue, I take the stance of wrye himself: I only have so much time. Forget it.
So do we all divide factions for bash?... 
On the TES side... Morrowind support is still missing, but is `the` fav chapter of TES players averaged overall.
Take it enjoyingly. Everyone likes to play the new game every once and a while. I don't blame ya. :)

**Comment by Sharlikran** [comment]
I see the changes were added to dev from Lojack's branch.  However, I am supposed to rebase off of 151-skyrim-patchers aren't I?  If so I can't really Rebase off of dev.

**Comment by Utumno** [comment]
You just pull upstream dev then issue `git rebase --interactive --onto dev dev feature-sharlikran-valda-fallout` - edit out the fallout 4 commits to avoid any unneeded conflicts - squash split etc - then just go to the tree checkout the branches that are left behind and right click on the new commits and choose reset so and so to this commit > hard. As always - what bothers you ?

**Comment by Sharlikran** [comment]
I have a lot more branches then that to deal with.  That still helps get me going, thanks.

**Comment by Utumno** [comment]
When you have branches like 

branch1 - branch2 - branch3

one way is to rebase branch3 and then force reset (in gitk) the other ones - I do it all the time

**Comment by Sharlikran** [comment]
I have too many other changes to feel comfortable doing that.  However, I'm just going to add the last change I need from Lojack's branch and then wait until you rebase 151-skyrim-patchers off of dev.  Besides, I'm fine with how things are for now.  I'm sure you are busy so whenever you have time to update 151-skyrim-patchers is fine but I'm going to wait to do a full rebase until that is done.

I'm able to play Fallout 4 now on my older LGA 775 it's just slow so I'm just having too much fun anyway.

**Comment by Utumno** [comment]
Hahaha - good playthrough

I pushed a rebase of your feature branch here: https://github.com/wrye-bash/wrye-bash/tree/feature-sharlikran-valda-fallout

Pull it and use interactive rebase to clean it up a bit - like squashing the white space commit etc - see you laterzzzzz

**Comment by Sharlikran** [comment]
How do I use GitK to compare my origin branch with the upstream branch?

**Comment by Utumno** [comment]
git fetch upstream feature-sharlikran-valda-fallout
git checkout -b FETCH_HEAD upstream-feature-sharlikran-valda-fallout

;)

**Comment by InsanePlumber** [comment]
[url=http://www.nexusmods.com/fallout4/mods/4421/]ArchiveInvalidation[/url]

**Comment by Utumno** [comment]
> ArchiveInvalidation
http://www.nexusmods.com/fallout4/mods/4421/?

A tweak anyone (just verify before)

**Comment by lojack5** [comment]
That's just an alternative to what we already provide, the INI tweaks he says are unnecessary (the do the same thing in different ways).

**Comment by lojack5** [comment]
I will however add this to the INI Tweaks, possibly marking it as the recommended one too since it looks like this way has advantages.

To set someone else as author on a commit I just need the Github email right @Utumno ?

**Comment by Utumno** [comment]
I guess yes actually never tested - just see how your commits look like in gitk and imitate that - probably you have to `git commit --amend --author` whatever the syntax is

**Comment by Utumno** [comment]
Hmm - apparently the re is an error on first running the ini tab - IniInfos is not populated ? See: http://forums.bethsoft.com/topic/1558694-wipz-wrye-flashexp-fallout-4-installer-update/page-2#entry24706480

```
wxPython: stdout/stderr
Traceback (most recent call last):
  File "bash\basher\__init__.pyo", line 660, in OnLeftDown
  File "bash\bolt.pyo", line 1219, in __getitem__
KeyError: 0
```

User deleted both the Fallout4.ini and the Fallout4prefs.ini and can reliably reproduce. Maybe an ini tabs issue in general - not a showstopper but we must fix: http://forums.bethsoft.com/topic/1558694-wipz-wrye-flashexp-fallout-4-installer-update/page-2#entry24710050

@lojack5: I don't have fallout - although as I said seems like an edge case in ini tab. Please test when you come round to it

**Comment by Sharlikran** [comment]
@lojack5 also when you have time could you advise me on how to update the valid INI sections?  The reason I ask is there is a function to look and see if errors will result when the tweak is applied.  Whether or not it's a rudimentary verification it's reporting a false error in regards to Fallout 4 and I'd like to move the sets or tuples to constants.py as mentioned [in this post](http://forums.bethsoft.com/topic/1558694-wipz-wrye-flashexp-fallout-4-installer-update/?p=24706722).

**Comment by Utumno** [comment]
Closing this - we do support the game

**Comment by Infernio** [comment]
Ended up being duplicated by and implemented in #251.
